### PR TITLE
[testing] acceptance test for several resources

### PR DESCRIPTION
**Pull request by ryanking**
Working towards having acceptance tests for all resources, this adds for table_grant, stage_grant, resource_monitor, account_grant and warehouse_grant.

## Test Plan

* [x] acceptance tests

* [x] …

## References

*

### CODE REVIEW

File: pkg/resources/account_grant_acceptance_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccAccountGrant_defaults(t *testing.T) {
+	roleName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+
+	resource.Test(t, resource.TestCase{
```
Comment (cziie) [inline pkg/resources/account_grant_acceptance_test.go:14]: **[tfproviderlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>AT001: missing CheckDestroy

File: pkg/resources/resource_monitor_grant_acceptance_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccResourceMonitor_defaults(t *testing.T) {
+	wName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+	roleName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+
+	resource.Test(t, resource.TestCase{
```
Comment (cziie) [inline pkg/resources/resource_monitor_grant_acceptance_test.go:15]: **[tfproviderlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>AT001: missing CheckDestroy

File: pkg/resources/stage_grant_acceptance_test.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccStageGrant_defaults(t *testing.T) {
+	name := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+
+	resource.Test(t, resource.TestCase{
```
Comment (cziie) [inline pkg/resources/stage_grant_acceptance_test.go:14]: **[tfproviderlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>AT001: missing CheckDestroy

File: pkg/resources/table_grant_acceptance_test.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccTableGrant_defaults(t *testing.T) {
+	name := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+
+	resource.Test(t, resource.TestCase{
```
Comment (cziie) [inline pkg/resources/table_grant_acceptance_test.go:14]: **[tfproviderlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>AT001: missing CheckDestroy

File: pkg/resources/account_grant_acceptance_test.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+		Providers: providers(),
+		Steps: []resource.TestStep{
+			{
+				Config: accountGrantConfig(roleName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_account_grant.test", "privilege", "MONITOR USAGE"),
```
Comment (edulop91) [inline pkg/resources/account_grant_acceptance_test.go:20]:
```suggestion
					resource.TestCheckResourceAttr("snowflake_account_grant.test", "privilege", privilegeMonitorUsage.String()),
```

File: pkg/resources/resource_monitor_grant_acceptance_test.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+		Providers: providers(),
+		Steps: []resource.TestStep{
+			{
+				Config: resourceMonitorGrantConfig(wName, roleName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_resource_monitor_grant.test", "monitor_name", wName),
+					resource.TestCheckResourceAttr("snowflake_resource_monitor_grant.test", "privilege", "MONITOR"),
```
Comment (edulop91) [inline pkg/resources/resource_monitor_grant_acceptance_test.go:22]:
```suggestion
					resource.TestCheckResourceAttr("snowflake_resource_monitor_grant.test", "privilege", privilegeMonitor.String()),
```

**Review by edulop91 — APPROVED** [review]

File: pkg/resources/account_grant_acceptance_test.go [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+func TestAccAccountGrant_defaults(t *testing.T) {
+	roleName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+
+	resource.Test(t, resource.TestCase{
+		Providers: providers(),
+		Steps: []resource.TestStep{
+			{
+				Config: accountGrantConfig(roleName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_account_grant.test", "privilege", privilegeMonitorUsage.String()),
```
Comment (cziie) [inline pkg/resources/account_grant_acceptance_test.go:20]: **[tfproviderlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>undeclared name: privilegeMonitorUsage

**Pull request merged**